source/developers/in-context-editing-ftl.rst

```diff
@@ -0,0 +1,368 @@
+
+    .. code-block:: guess
+
+	    <#import "/templates/system/common/cstudio-support.ftl" as studio/>
+
+|
+
+At the bottom of your template insert the following: (Note the example shows a traditional HTML page however other formats/levels of granularity are supported
+
+    .. code-block:: guess
+
+	        <@studio.toolSupport/>
+	      </body>
+	    </html>
+
+|
+
+--------------------------
+In-Context Editing Pencils
+--------------------------
+
+In context editing renders pencils on the screen that invoke editing controls when clicked.  This allows authors to quickly/visually identify editable content and make changes.
+
+.. image:: /_static/images/ice-example.png
+        :align: center
+        :width: 70 %
+        :alt: In context editing example
+
+|
+
+To enable in-context editing simply add the following attribute to the container/element where you want to place the editing control
+
+    .. code-block:: guess
+
+	    <@studio.iceAttr iceGroup="author"/>
+
+|
+
+For embedded components, to enable in-context editing simply add the following attribute and tag attribute to the embedded component template
+
+    .. code-block:: guess
+
+	    <@studio.iceAttr component=contentModel/>
+
+|
+
+Tag Attributes
+--------------
+
++----------------+------------------------------------+-------------------------------------------+
+| Attribute Name | Required                           | Expected Value                            |
++================+====================================+===========================================+
+|| iceGroup      || No (unless path is not supplied)  || the label/id assigned to iceGroup on     |
+||               ||                                   || fields in your content model.            |
++----------------+------------------------------------+-------------------------------------------+
+|| path          || No                                || the path of the item. This is typically  |
+||               || (unless iceGroup is not supplied) || just mode.storeUrl.                      |
+||               ||                                   ||                                          |
+||               ||                                   || If path is not supplied the system       |
+||               ||                                   || will assume the outermost object e.g.    |
+||               ||                                   || the page as the path.                    |
++----------------+------------------------------------+-------------------------------------------+
+|| label         || No (but it's a best practice)     || UI will use label if it exists. Otherwise|
+||               ||                                   || the iceGroup or path will be used.       |
++----------------+------------------------------------+-------------------------------------------+
+|| component     || No                                || a |SiteItem| object                      |
```

Comment:
It's not required only for backwards compatibility, maybe we should say it is required?